app/controllers/events_controller.rb

```diff
@@ -4,12 +4,14 @@ class EventsController < ApplicationController
   before_action :is_logged_in?, only: [:student, :coach]
 
   def index
-    events = [ Workshop.past.all ]
-    events << Course.past.all
-    events << Meeting.past.all
-    events << Event.past.all
-    events = events.compact.flatten.sort_by(&:date_and_time).reverse.group_by(&:date)
-    @past_events = events.map.inject({}) { |hash, (key, value)| hash[key] = EventPresenter.decorate_collection(value); hash}
+    record_limit = 30
+    events = [ Workshop.past_display(record_limit) ]
+    events << Course.past_display(record_limit)
+    events << Meeting.past_display(record_limit)
+    events << Event.past_display(record_limit)
+    events = events.compact.flatten.sort_by(&:date_and_time).reverse[0...record_limit].group_by(&:date)
```

Comment:
I 100% agree - to rephrase -
Let's say we want to return 40 (for easier maths)

I think it's better that we return 10 of each, and then sort everything, and then output everything
rather than
returning 40 of each, sorting everything, and then outputting the first 40 (which is what this currently does)